hikari/impl/event_manager_base.py

```diff
@@ -289,26 +283,41 @@ def filtered(
     else:
         event_types = event_types.dispatches()
 
-    def decorator(method: UnboundMethodT[EventManagerBaseT], /) -> UnboundMethodT[EventManagerBaseT]:
+    bitmask = 0
+    for event_type in event_types:
+        bitmask |= event_type.bitmask()
+
+    # https://github.com/python/mypy/issues/2087
+    def decorator(method: UnboundMethodT[EventManagerBaseT], /) -> _FilteredMethodT:
```

Comment:
```suggestion
    def decorator(method: typing.Any, /) -> _FilteredMethodT:
```

Would this work and remove all the ignores?

Reply:
Both of these changes kinda breaks the typing though; the methods being typed as _FilteredMethodT on the class is wrong since you'd also have to explicitly pass self and do you really want all the methods on the impl to be typed as _FilteredMethodT which includes internal detail which isn't meant to be exposed typing wise